### PR TITLE
Fix new Mac warnings

### PR DESCRIPTION
**Pull request by Atlante45**

### CODE REVIEW

**Comment by Atlante45** [comment]
## Test plan:
No functional changes
- Smoke test Interface by launching it on Windows and Mac.

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/13539/HighFidelity-Beta-PR14125-ec659d1.apk

File: libraries/gpu-gl-common/src/gpu/gl/GLShared.h [hunk 1]
```diff
@@ -19,7 +19,11 @@ Q_DECLARE_LOGGING_CATEGORY(gpugllogging)
 Q_DECLARE_LOGGING_CATEGORY(trace_render_gpu_gl)
 Q_DECLARE_LOGGING_CATEGORY(trace_render_gpu_gl_detail)
 
+#if defined(Q_OS_MAC)
```
Comment (sethalves) [inline libraries/gpu-gl-common/src/gpu/gl/GLShared.h]: I get a warning about this on Linux also.  let's make this #if clang

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/13542/HighFidelity-Beta-PR14125-c25de57.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/13542/HighFidelity-Beta-PR14125-c25de57.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/13542/HighFidelity-Beta-Interface-PR14125-c25de57.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/13542/HighFidelity-Beta-PR14125-c25de57.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/13542/HighFidelity-Beta-Interface-PR14125-c25de57.exe

**Comment by Atlante45** [comment]
## Test Plan:
No functional changes, only fixing compiler warnings, so this is a simple smoke test.
- Run a sandbox and an Interface
- Connect to the Sandbox and make sure you can move around and create/delete entities

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/13753/HighFidelity-Beta-PR14125-0634efb.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/13753/HighFidelity-Beta-Interface-PR14125-0634efb.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/13753/HighFidelity-Beta-PR14125-0634efb.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/13753/HighFidelity-Beta-Interface-PR14125-0634efb.exe

**Comment by antoninahifi** [comment]
currently testing

**Comment by antoninahifi** [comment]
Build this please

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/13955/HighFidelity-Beta-PR14125-0634efb.apk

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/13955/HighFidelity-Beta-PR14125-0634efb.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/13955/HighFidelity-Beta-Interface-PR14125-0634efb.exe

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/13955/HighFidelity-Beta-PR14125-0634efb.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/13955/HighFidelity-Beta-Interface-PR14125-0634efb.dmg

**Comment by QliCknox** [comment]
Currently Testing

**Comment by antoninahifi** [comment]
PASS @conklin94122

**Pull request merged**